src/core/file_sys/registered_cache.cpp

```diff
@@ -547,6 +547,62 @@ InstallResult RegisteredCache::InstallEntry(const XCI& xci, bool overwrite_if_ex
     return InstallEntry(*xci.GetSecurePartitionNSP(), overwrite_if_exists, copy);
 }
 
+bool RegisteredCache::RemoveExistingEntry(const u64 title_id) {
+    const auto delete_nca = [this](const NcaID& id) {
+        const auto path = GetRelativePathFromNcaID(id, false, true, false);
+
+        if (dir->GetFileRelative(path) == nullptr) {
+            return false;
+        }
+
+        Core::Crypto::SHA256Hash hash{};
+        mbedtls_sha256_ret(id.data(), id.size(), hash.data(), 0);
+        const auto dirname = fmt::format("000000{:02X}", hash[0]);
+
+        const auto dir2 = GetOrCreateDirectoryRelative(dir, dirname);
+
+        const auto res = dir2->DeleteFile(fmt::format("{}.nca", Common::HexToString(id, false)));
+
+        return res;
+    };
+
+    // Get the Content Provider
+    const auto& installed = Core::System::GetInstance().GetContentProvider();
+    // If an update exists, remove
+    if (installed.HasEntry(title_id, ContentRecordType::Meta)) {
+        LOG_INFO(Loader,
+                 "Previous Update (v{}) for title_id={:016X} detected! Attempting to remove...",
+                 installed.GetEntryVersion(title_id).value_or(0), title_id);
+        // Get all the ncas associated with the current update CNMT and delete them
+        const auto& meta_old_id =
```

Comment:
```suggestion
        const auto meta_old_id =
```

`GetNcaIDFromMetadata` returns by value. Ditto for all other occurrences below.